scipy/optimize/tests/test_quadratic_assignment.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_array_equal
+from scipy.optimize import quadratic_assignment
+
+
+@pytest.mark.parametrize("n", [10, 100])
+@pytest.mark.parametrize("directed", [False, True])
+def test_weighted_graph_isomorphism(n, directed):
+    rng = np.random.RandomState(seed=0)
+    A = np.abs(rng.randn(n, n))
+    if not directed:
+        A = (A + A.T) / 2
+
+    P = rng.permutation(n)
+    B = A[P][:, P]
+
+    result = quadratic_assignment(A, B, maximize=True)
+    assert_array_equal(result.col_ind, P)
+
+
+def test_directed_weighted_graph_matching():
+    # Example in section IVB of Umeyama:
+    # "An eigendecomposition approach to weighted graph matching"
+    A = np.array([[0, 3, 4, 2],
+                  [0, 0, 1, 2],
+                  [1, 0, 0, 1],
+                  [0, 0, 1, 0]])
+
+    B = np.array([[0, 4, 2, 4],
+                  [0, 0, 1, 0],
+                  [0, 2, 0, 2],
+                  [0, 1, 2, 0]])
+
+    result = quadratic_assignment(A, B, maximize=True)
+    assert_array_equal(result.col_ind, [0, 2, 3, 1])
```

Comment:
```suggestion
    assert_array_equal(result.col_ind, [0, 3, 1, 2])
```
This is not the answer printed in Umeyama.
![image](https://user-images.githubusercontent.com/6570539/89611066-08271e80-d831-11ea-83c1-db45116986f8.png)

Reply:
@asaadeldin11 @jovo Do you have easy acess to an implementation of Umeyama's method? Can you check the results on this problem?

Reply:
I think Umeyama would recommend the transpose of the permutation we were using.

Reply:
> This is not the answer printed in Umeyama.

If we do:
```
>>> result = linear_sum_assignment(W, maximize=maximize)
>>> P = np.zeros(W.shape, dtype=int)
>>> P[result] = 1
>>> P
array([[1 0 0 0]
       [0 0 1 0]
       [0 0 0 1]
       [0 1 0 0]])
```

Since we have `row_ind, col_ind = result` I think this is consistent with Umeyama? Note that I am in agreement that Umeyama's permutation is transposed with respect to that of FAQ.

Reply:
Sorry, I edited my post. It was old.
It _is_ in agreement with Umeyama, but Umeyama solves the transposed problem (and for that, Umeyema notes that it is the global optimum). You'll find that it does not do as well for our problem as the transpose does (and you'll find that for our problem, this transpose is the global optimum).

Reply:
OK - we're on the same page.
I have changed the permutation convention. Also the superficial one you mention.